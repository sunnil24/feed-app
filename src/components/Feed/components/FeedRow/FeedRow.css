.colored {
  background-color: #e6e6df;
}

.feed-wrapper {
  align-items: center;
  color: #828282;
  display: flex;
}

.feed-comment-count {
  flex: 0 0 35px;
  text-align: right;
  padding-right: 4px;
}

.feed-upvote-count {
  flex: 0 0 35px;
  text-align: right;
  padding: 0 4px;
}

.feed-upvote-cta {
  flex: 0 0 20px;
  font-size: 10px;
}

@media only screen and (max-width: 600px) {
  .feed-wrapper {
    flex-wrap: wrap;
  }

  .feed-title {
    padding: 10px 0 0;
    flex-basis: 100%;
    line-height: 20px;
  }

  .feed-row {
    padding: 16px 0;
  }

  .feed-title {
    flex-basis: auto;
  }

  .feed-comment-count {
    text-align: left;
    flex-basis: 100px;
    padding: 0;
  }

  .feed-upvote-count {
    text-align: left;
    padding: 0;
    flex-basis: 100px;
  }

  .feed-upvote-count:before {
    content: "Vote: ";
    font-weight: bold;
  }

  .feed-comment-count:before {
    content: "Comment: ";
    font-weight: bold;
  }

  .feed-upvote-cta {
    flex-basis: 100px;
    font-size: 12px;
  }

  .feed-upvote-cta:before {
    content: "Upvote: ";
    font-weight: bold;
    color: #000;
  }

  .primary-text.pagination {
    margin-left: 5%;
  }
  .feed-title {
    flex-basis: auto;
  }

  .feed-comment-count {
    text-align: left;
    flex-basis: 100px;
    padding: 0;
  }

  .feed-upvote-count {
    text-align: left;
    padding: 0;
    flex-basis: 100px;
  }

  .feed-upvote-count:before {
    content: "Vote: ";
    font-weight: bold;
  }

  .feed-comment-count:before {
    content: "Comment: ";
    font-weight: bold;
  }

  .feed-upvote-cta {
    flex-basis: 100px;
    font-size: 12px;
  }

  .feed-upvote-cta:before {
    content: "Upvote: ";
    font-weight: bold;
    color: #000;
  }

  .primary-text.pagination {
    margin-left: 5%;
  }
}
