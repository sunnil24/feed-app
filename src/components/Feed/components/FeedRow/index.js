import React from "react";
import classnames from "classnames";
import moment from "moment";
import { Link } from "react-router-dom";

//styles
import "./FeedRow.css";

//components
import Main from "../../../Main";
import { getDomainName } from "../../../../utils/common";

const FeedRow = ({ className, colored, data }) => {
  const { title, url, num_comments, points, created_at, author } = data;
  debugger;
  return (
    <div
      className={classnames(
        "feed-row",
        {
          colored: colored,
        },
        className
      )}
    >
      <Main className="feed-wrapper">
        <div className="feed-comment-count dark-text">{num_comments}</div>
        <div className="feed-upvote-count dark-text">{points}</div>
        <div className="feed-upvote-cta">
          <Link onClick={() => {}} className="inherit-text">
            &#9650;
          </Link>
        </div>
        <div className="feed-title">
          <a href={url} className="dark-text">
            {title}
          </a>
          &nbsp; ({getDomainName(url)}) by{" "}
          <span className="dark-text">{author} </span>
          {moment(created_at).fromNow()} [
          <Link onClick={() => {}} className="dark-text">
            hide
          </Link>
          ]
        </div>
      </Main>
    </div>
  );
};

FeedRow.defaultProps = {
  colored: false,
};

export default FeedRow;
