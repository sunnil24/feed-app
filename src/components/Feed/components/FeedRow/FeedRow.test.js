import React from "react";
import { shallow } from "enzyme";

import FeedRow from "./index";

const data = {
  title: "Some title",
  url: "http://some-utl.com",
  num_comments: 222,
  points: 10,
  created_at: 67564652,
  author: "Sunil",
};

const props = {
  data,
  className: "some-class",
  colored: true,
};

describe("<FeedRow />", () => {
  const wrapped = shallow(<FeedRow {...props} />);

  it("should render the Title Component correctly", () => {
    expect(wrapped).toMatchSnapshot();
  });
});
