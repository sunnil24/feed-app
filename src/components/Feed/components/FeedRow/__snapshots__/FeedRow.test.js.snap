// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeedRow /> should render the Title Component correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FeedRow
    className="some-class"
    colored={true}
    data={
      Object {
        "author": "Sunil",
        "created_at": 67564652,
        "num_comments": 222,
        "points": 10,
        "title": "Some title",
        "url": "http://some-utl.com",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Main
        className="feed-wrapper"
      >
        <div
          className="feed-comment-count dark-text"
        >
          222
        </div>
        <div
          className="feed-upvote-count dark-text"
        >
          10
        </div>
        <div
          className="feed-upvote-cta"
        >
          <ForwardRef
            className="inherit-text"
            onClick={[Function]}
          >
            ▲
          </ForwardRef>
        </div>
        <div
          className="feed-title"
        >
          <a
            className="dark-text"
            href="http://some-utl.com"
          >
            Some title
          </a>
            (
          some-utl.com
          ) by
           
          <span
            className="dark-text"
          >
            Sunil
             
          </span>
          50 years ago
           [
          <ForwardRef
            className="dark-text"
            onClick={[Function]}
          >
            hide
          </ForwardRef>
          ]
        </div>
      </Main>,
      "className": "feed-row colored some-class",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            className="feed-comment-count dark-text"
          >
            222
          </div>,
          <div
            className="feed-upvote-count dark-text"
          >
            10
          </div>,
          <div
            className="feed-upvote-cta"
          >
            <ForwardRef
              className="inherit-text"
              onClick={[Function]}
            >
              ▲
            </ForwardRef>
          </div>,
          <div
            className="feed-title"
          >
            <a
              className="dark-text"
              href="http://some-utl.com"
            >
              Some title
            </a>
              (
            some-utl.com
            ) by
             
            <span
              className="dark-text"
            >
              Sunil
               
            </span>
            50 years ago
             [
            <ForwardRef
              className="dark-text"
              onClick={[Function]}
            >
              hide
            </ForwardRef>
            ]
          </div>,
        ],
        "className": "feed-wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 222,
            "className": "feed-comment-count dark-text",
          },
          "ref": null,
          "rendered": 222,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 10,
            "className": "feed-upvote-count dark-text",
          },
          "ref": null,
          "rendered": 10,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ForwardRef
              className="inherit-text"
              onClick={[Function]}
            >
              ▲
            </ForwardRef>,
            "className": "feed-upvote-cta",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "▲",
              "className": "inherit-text",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "▲",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "displayName": "Link",
              "propTypes": Object {
                "innerRef": [Function],
                "onClick": [Function],
                "replace": [Function],
                "target": [Function],
                "to": [Function],
              },
              "render": [Function],
            },
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <a
                className="dark-text"
                href="http://some-utl.com"
              >
                Some title
              </a>,
              "  (",
              "some-utl.com",
              ") by",
              " ",
              <span
                className="dark-text"
              >
                Sunil
                 
              </span>,
              "50 years ago",
              " [",
              <ForwardRef
                className="dark-text"
                onClick={[Function]}
              >
                hide
              </ForwardRef>,
              "]",
            ],
            "className": "feed-title",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Some title",
                "className": "dark-text",
                "href": "http://some-utl.com",
              },
              "ref": null,
              "rendered": "Some title",
              "type": "a",
            },
            "  (",
            "some-utl.com",
            ") by",
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Sunil",
                  " ",
                ],
                "className": "dark-text",
              },
              "ref": null,
              "rendered": Array [
                "Sunil",
                " ",
              ],
              "type": "span",
            },
            "50 years ago",
            " [",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "hide",
                "className": "dark-text",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "hide",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "displayName": "Link",
                "propTypes": Object {
                  "innerRef": [Function],
                  "onClick": [Function],
                  "replace": [Function],
                  "target": [Function],
                  "to": [Function],
                },
                "render": [Function],
              },
            },
            "]",
          ],
          "type": "div",
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Main
          className="feed-wrapper"
        >
          <div
            className="feed-comment-count dark-text"
          >
            222
          </div>
          <div
            className="feed-upvote-count dark-text"
          >
            10
          </div>
          <div
            className="feed-upvote-cta"
          >
            <ForwardRef
              className="inherit-text"
              onClick={[Function]}
            >
              ▲
            </ForwardRef>
          </div>
          <div
            className="feed-title"
          >
            <a
              className="dark-text"
              href="http://some-utl.com"
            >
              Some title
            </a>
              (
            some-utl.com
            ) by
             
            <span
              className="dark-text"
            >
              Sunil
               
            </span>
            50 years ago
             [
            <ForwardRef
              className="dark-text"
              onClick={[Function]}
            >
              hide
            </ForwardRef>
            ]
          </div>
        </Main>,
        "className": "feed-row colored some-class",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <div
              className="feed-comment-count dark-text"
            >
              222
            </div>,
            <div
              className="feed-upvote-count dark-text"
            >
              10
            </div>,
            <div
              className="feed-upvote-cta"
            >
              <ForwardRef
                className="inherit-text"
                onClick={[Function]}
              >
                ▲
              </ForwardRef>
            </div>,
            <div
              className="feed-title"
            >
              <a
                className="dark-text"
                href="http://some-utl.com"
              >
                Some title
              </a>
                (
              some-utl.com
              ) by
               
              <span
                className="dark-text"
              >
                Sunil
                 
              </span>
              50 years ago
               [
              <ForwardRef
                className="dark-text"
                onClick={[Function]}
              >
                hide
              </ForwardRef>
              ]
            </div>,
          ],
          "className": "feed-wrapper",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 222,
              "className": "feed-comment-count dark-text",
            },
            "ref": null,
            "rendered": 222,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 10,
              "className": "feed-upvote-count dark-text",
            },
            "ref": null,
            "rendered": 10,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ForwardRef
                className="inherit-text"
                onClick={[Function]}
              >
                ▲
              </ForwardRef>,
              "className": "feed-upvote-cta",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "▲",
                "className": "inherit-text",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "▲",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "displayName": "Link",
                "propTypes": Object {
                  "innerRef": [Function],
                  "onClick": [Function],
                  "replace": [Function],
                  "target": [Function],
                  "to": [Function],
                },
                "render": [Function],
              },
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <a
                  className="dark-text"
                  href="http://some-utl.com"
                >
                  Some title
                </a>,
                "  (",
                "some-utl.com",
                ") by",
                " ",
                <span
                  className="dark-text"
                >
                  Sunil
                   
                </span>,
                "50 years ago",
                " [",
                <ForwardRef
                  className="dark-text"
                  onClick={[Function]}
                >
                  hide
                </ForwardRef>,
                "]",
              ],
              "className": "feed-title",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Some title",
                  "className": "dark-text",
                  "href": "http://some-utl.com",
                },
                "ref": null,
                "rendered": "Some title",
                "type": "a",
              },
              "  (",
              "some-utl.com",
              ") by",
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Sunil",
                    " ",
                  ],
                  "className": "dark-text",
                },
                "ref": null,
                "rendered": Array [
                  "Sunil",
                  " ",
                ],
                "type": "span",
              },
              "50 years ago",
              " [",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "hide",
                  "className": "dark-text",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "hide",
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "displayName": "Link",
                  "propTypes": Object {
                    "innerRef": [Function],
                    "onClick": [Function],
                    "replace": [Function],
                    "target": [Function],
                    "to": [Function],
                  },
                  "render": [Function],
                },
              },
              "]",
            ],
            "type": "div",
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
